File: 1524-number-of-sub-arrays-with-odd-sum/1524-number-of-sub-arrays-with-odd-sum.java
class Solution {
    public int numOfSubarrays(int[] arr) {
        long oddCount = 0;
        long sum = 0;

        int mod = 1_000_000_007;

        for(int num : arr) {
            sum += num;
            oddCount += sum % 2;
        }

        oddCount += (arr.length - oddCount) * oddCount;

        return (int) (oddCount % mod);
    }
}

File: 1524-number-of-sub-arrays-with-odd-sum/1524-number-of-sub-arrays-with-odd-sum.c
int numOfSubarrays(int* arr, int arrSize) {
        long long oddCount = 0;
        long long sum = 0;

        int mod = 1000000007;

        for(int i = 0; i < arrSize; i++) {
            sum += arr[i];
            oddCount += sum % 2;
        }

        oddCount += (arrSize - oddCount) * oddCount;

        return (int) (oddCount % mod);
}